Spearhead Blocks refactored Initial Commit

onuro opened the pull request:
Refactored commit of Spearhead Blocks, now being a clone of Block Canvas, instead of child of Blockbase. Also had to re-commit due to folder name of the theme has been renamed to match the naming conventions(maybe I shouldn't have created another PR, please let me know)

I tried the dark mode auto switch via style but I think I'm for now lacking the skill to reflect the dark mode changes to the editor. Ideas welcome :)

Thanks!

onuro committed:
Spearhead Blocks refactored Initial Commit

Refactored commit of Spearhead Blocks, now being a clone of Block Canvas, instead of child of Blockbase

pbking commented on spearhead-blocks/parts/header.html:
```diff
@@ -0,0 +1,19 @@
+<!-- /wp:group --></div>
+<!-- /wp:group -->
+
+<!-- wp:navigation {"ref":14,"__unstableLocation":"primary","layout":{"type":"flex","setCascadingProperties":true,"justifyContent":"right","orientation":"horizontal"},"style":{"spacing":{"margin":{"top":"0"}}}} /--></div>
```
The navigation block has a reference.  This should be removed.

pbking commented on spearhead-blocks/style.css:
```diff
@@ -0,0 +1,121 @@
+/*
+ * Dark Mode
+ */
+@media (prefers-color-scheme: dark) {
+  body {
```
To get this to work in the editor you'll need to use:
```
@media (prefers-color-scheme: dark) {
  body.editor-styles-wrapper,
  body {
```

onuro replied:
Worked like charm ✨ 🙏🏿

pbking commented on spearhead-blocks/theme.json:
```diff
@@ -0,0 +1,455 @@
+					"name": "Background",
+					"slug": "background"
+				},
+				{
+					"color": "#fafbf6",
+					"name": "Tertiary",
+					"slug": "tertiary"
+				},
```
I don't see the tertiary slug being used anywhere.  Is it necessary to include?

onuro replied:
Ah, thought original Spearhead also had it. Apparently not :) Removing.

onuro committed:
Review updates

- removal of ref id from header template
- dark mode editor style added
- unused tertiary color removed from the palette
- screenshot added

mikachan commented on spearhead-blocks/theme.json:
```diff
@@ -0,0 +1,450 @@
+							}
+						},
+						"color": {
+							"text": "var:preset|color|foreground"
```
It looks like this was originally set to the primary color, based on the Seedlet base CSS [here](https://github.com/Automattic/themes/blob/trunk/seedlet/assets/sass/base/_base.scss#L3).

If we switched the text color for primary we could perhaps use foreground for the hover color.

mikachan commented on spearhead-blocks/theme.json:
```diff
@@ -0,0 +1,450 @@
+							"text": "var:preset|color|primary"
+						},
+						"typography": {
+							"textDecoration": "none"
```
Based on the original theme, it looks like `"textDecoration": "none"` needs to be applied to the :hover and :focus styles too.

mikachan commented on spearhead-blocks/theme.json:
```diff
@@ -0,0 +1,450 @@
+				"typography": {
+					"fontSize": "var:preset|font-size|medium",
+					"fontStyle": "normal",
+					"fontWeight": "700",
```
The font-weight for post titles was set to 500 on the original theme.

onuro replied:
Thanks but when I tried that, it wasn't matching with the demo, where all post titles were also 700 in weight:
https://spearheaddemo.wordpress.com/

mikachan commented on spearhead-blocks/theme.json:
```diff
@@ -0,0 +1,450 @@
+			}
+		},
+		"spacing": {
+			"blockGap": "35px"
```
I can see a few gaps in the original theme using 0.5em (I think it's defaulting to the GB default gap spacing), rather than 35px.

mikachan commented on spearhead-blocks/parts/footer.html:
```diff
@@ -0,0 +1,11 @@
+	<!-- wp:group {"style":{"spacing":{"padding":{"top":"80px","bottom":"30px"}}}} -->
+	<div class="wp-block-group" style="padding-top:80px;padding-bottom:30px">
+		<!-- wp:paragraph {"align":"center"} -->
+		<p class="has-text-align-center">Proudly Powered by <a href="https://wordpress.org" rel="nofollow">WordPress</a></p>
```
The original theme includes the site title in the footer, next to the 'powered by WordPress' line:

<img width="447" alt="image" src="https://user-images.githubusercontent.com/1645628/187716796-c8387860-de7d-4caf-bf1a-e7b6a22b1227.png">

mikachan reviewed (commented):
This is looking really good! I've left some inline comments, which are mainly comparing this theme to the original Spearhead theme. I'm not sure we want to match all of the differences exactly - I just wanted to point out the ones I noticed.

We should probably close https://github.com/Automattic/themes/pull/6331 in favour of this PR, if you're happy moving ahead with this one.

onuro committed:
Fixes to Theme.json and Footer Template

This commit fixes the issues mentioned in the PR

- Footer template now has the site title in it
- Block spacing and font weight fixes
